pkg/xray/reports.go

```diff
@@ -0,0 +1,802 @@
+package xray
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"github.com/jfrog/terraform-provider-shared/validator"
+)
+
+var getReportSchema = func(filtersSchema map[string]*schema.Schema) map[string]*schema.Schema {
+	return util.MergeMaps(
+		getProjectKeySchema(false, ""),
+		map[string]*schema.Schema{
+			"report_id": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				Computed:         true,
+				Description:      "Report ID",
+				ValidateDiagFunc: validator.StringIsNotEmpty,
+			},
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				Description:      "Name of the report.",
+				ValidateDiagFunc: validator.StringIsNotEmpty,
+			},
+			"resources": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "The list of resources to include into the report.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"repository": {
+							Type:        schema.TypeSet,
+							Optional:    true,
+							MinItems:    1,
+							Description: "The list of repositories for the report. Only one type of resource can be set per report.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "Repository name.",
+									},
+									"include_path_patterns": {
+										Type:        schema.TypeSet,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+										Set:         schema.HashString,
+										Optional:    true,
+										Description: "Include path patterns.",
```

Comment:
IIRC, validation func works in nested schema too. So let's apply, at min, not empty validation here (and the other similar attributes).

Reply:
Added for this field.